app/models/address.rb

```diff
@@ -25,4 +33,22 @@ def to_a
       postcode
     ].map(&:presence).compact
   end
+
+  private
+
+  def set_point_from_postcode
+    return if postcode.blank? || point.present?
+    geocode = geocode_postcode
+    if geocode && geocode.data['address_components'].any? { |a| a['types'] == ['postal_code'] }
+      self.point = {
+        type: 'Point',
+        coordinates: [geocode.longitude, geocode.latitude]
+      }
+    end
+  end
+
+  def geocode_postcode
+    results = Geocoder.search(postcode)
+    results.count == 1 ? results.first : nil
```

Comment:
I'm not usually a fan of `try` but this line might be clarified by `results.try(:first)`?